moabb/pipelines/deep_learning.py

```diff
@@ -0,0 +1,749 @@
+"""
+Deep learning that work on Moabb.
+Implementation using the tensorflow, keras and scikeras framework.
+"""
+
+# Authors: Igor Carrara <igor.carrara@inria.fr>
+#          Bruno Aristimunha <b.aristimunha@gmail.com>
+#          Sylvain Chevallier <sylvain.chevallier@universite-paris-saclay.fr>
+
+# License: BSD (3-clause)
+
+from typing import Any, Dict
+
+import tensorflow as tf
+from keras import backend as K
+from keras.constraints import max_norm
+from keras.layers import (
+    Activation,
+    Add,
+    AveragePooling2D,
+    AvgPool2D,
+    Concatenate,
+    Conv2D,
+    Dense,
+    DepthwiseConv2D,
+    Dropout,
+    Flatten,
+    Input,
+    Lambda,
+    LayerNormalization,
+    MaxPooling2D,
+)
+from keras.layers.normalization.batch_normalization import BatchNormalization
+from keras.models import Model, Sequential
+from scikeras.wrappers import KerasClassifier
+
+from moabb.pipelines.utils_deep_model import EEGNet, TCN_block
+
+
+# =====================================================================================
+# ShallowConvNet
+# =====================================================================================
+def square(x):
+    """
+    Function to square the input tensor element-wise.
+    Element-wise square.
+    """
+    return K.square(x)
+
+
+def log(x):
+    """
+    Function to take the log of the input tensor element-wise.
+    We use a clip to avoid taking the log of 0.
+    min_value=1e-7, max_value=10000
+    Parameters
+    ----------
+    x: tensor
+
+    Returns
+    -------
+    tensor
+    """
+    return K.log(K.clip(x, min_value=1e-7, max_value=10000))
+
+
+class KerasShallowConvNet(KerasClassifier):
+    """Keras implementation of the Shallow Convolutional Network as described
+    in Schirrmeister et. al. (2017), Human Brain Mapping.
+
+    This implementation is taken from code by the Army Research Laboratory (ARL)
+    at https://github.com/vlawhern/arl-eegmodels
+
+    We use the original parameter implemented on the paper.
+
+    Note that this implementation has not been verified by the original
+    authors. We do note that this implementation reproduces the results in the
+    original paper with minor deviations.
```

Comment:
Could you include the bibliographical reference in the documentation following this format? https://github.com/NeuroTechX/moabb/blob/develop/moabb/pipelines/classification.py#L20